sql/core/src/test/scala/org/apache/spark/sql/VariantSuite.scala

```diff
@@ -73,5 +73,12 @@ class VariantSuite extends QueryTest with SharedSparkSession {
       values.map(v => if (v == null) "null" else v.debugString()).sorted
     }
     assert(prepareAnswer(input) == prepareAnswer(result))
+
+    withTempDir { dir =>
```

Comment:
The `basic tests` test case also test parquet write an read, why it didn't expose the bug?

Reply:
Because the variant values it writes are all non-null. This only causes an issue when there is a null variant value.